Tests/Functional/Listener/FormRequestTest.php

```diff
@@ -0,0 +1,238 @@
+<?php
+
+
+namespace Functional\Listener;
+
+
+use Sulu\Bundle\FormBundle\Configuration\MailConfiguration;
+use Sulu\Bundle\FormBundle\Entity\Form;
+use Sulu\Bundle\FormBundle\Entity\FormField;
+use Sulu\Bundle\FormBundle\Entity\FormFieldTranslation;
+use Sulu\Bundle\FormBundle\Entity\FormTranslation;
+use Sulu\Bundle\FormBundle\Entity\FormTranslationReceiver;
+use Sulu\Bundle\TestBundle\Testing\SuluTestCase;
+
+class FormRequestTest extends SuluTestCase
+{
+
+    /**
+     * @var \Symfony\Bundle\FrameworkBundle\KernelBrowser
+     */
+    private $client;
+
+    /**
+     * @var \Doctrine\ORM\EntityManagerInterface
+     */
+    private $entityManager;
+
+    public function testPageRendersWithoutForm()
+    {
+        $this->inititalize();
+
+        $this->updateHomePage(null);
+
+        $this->client->request('GET', '/');
+//        $this->assertResponseIsSuccessful();
+        $response = $this->client->getResponse();
+        $this->assertHttpStatusCode(200, $response);
+    }
+
+    public function testPageRendersWithSetForm()
+    {
+        $this->inititalize();
+
+        $form = $this->createFullForm();
+        $this->updateHomePage($form);
+        $this->doSendForm($form);
+
+        $mailCollector = $this->client->getProfile()->getCollector('swiftmailer');
+        // 2 messages should be send 1 to admin and 1 to email
+        $this->assertSame(2, $mailCollector->getMessageCount());
+
+        $this->assertEmailCount(0);
+    }
+
+    public function testDoesNotMailWhenNullHandlerSet()
+    {
+        $_ENV['MAIL_HANDLER'] = 'NullHelper';
+        $this->inititalize();
+
+        $form = $this->createFullForm();
+        $this->updateHomePage($form);
+        $this->doSendForm($form);
+
+        $mailCollector = $this->client->getProfile()->getCollector('swiftmailer');
+        // 2 messages should be send 1 to admin and 1 to email
+        $this->assertSame(0, $mailCollector->getMessageCount());
+
+        $this->assertEmailCount(0);
+    }
+
+    public function testUsesMailerWhenMailerHandlerSet()
+    {
+        $_ENV['MAIL_HANDLER'] = 'MailerHelper';
```

Comment:
would not set a `$_ENV` here would create seperate TestCase for the `Mailer` and set there the KernelConfiguration `environment`: https://github.com/sulu/sulu/blob/3b41b627fbc1cf93be7f98eb3e3d4ebe35883b35/src/Sulu/Bundle/TestBundle/Testing/KernelTrait.php#L66-L69

to something like `test_mailer` and then in then create a `test_mailer/config.yaml` or `config_test_mailer.yaml` which is loaded by the test kernel where:

```
sulu_form:
     mail:
           helper: 'mailer'
```

is configured